akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcProjectionImpl.scala

```diff
@@ -674,10 +674,21 @@ private[projection] object R2dbcProjectionImpl {
               case Duplicate =>
                 Future.successful(None)
               case RejectedSeqNr =>
-                replayIfPossible(env, observer(context)).map(_ => None)(ExecutionContext.parasitic)
+                replayIfPossible(env, observer(context)).flatMap {
+                  case true => // replayed missing events, now accept the current envelope for processing downstream
+                    loadEnvelope(env, sourceProvider).map { loadedEnvelope =>
```

Comment:
...but then do we actually process the envelope?

Reply:
Yes, so that the envelope and its context are passed downstream for the save offsets after regular processing.